### PR TITLE
Feature/adapt sdk with new smart contract

### PR DESCRIPTION
**Pull request by Le-Caignec**
- [x] refactor new smart contract naming => `DataProtectorSharing`
- [x] remove `collectionTokenId` from params when calling smart contract
- [x] add `appWhitelist` param in `addToCollection` function
- [x] add new function `addAppToAppWhitelist`
- [x] add new function `createAppWhitelist`
- [x] add new function `getUserAppWhitelist`
- [x] add corresponding tests
- [x] change `subscriptionExpiration` for `lastSubscriptionExpiration`
- [x] change `rentalExpiration` for `lastRentalExpiration`
- [x] change `removeCollection` for `burn`(inheritance from ERC721 interface => less gas in SC)

### CODE REVIEW

File: packages/sdk/src/lib/dataProtectorSharing/consumeProtectedData.ts [hunk 1]
```diff
@@ -47,26 +56,18 @@ export const consumeProtectedData = async ({
...
-    if (appOwner !== sharingContractAddress) {
-      throw new WorkflowError(
-        'The app related to the protected data is not owned by the DataProtector Sharing contract'
-      );
-    }
-
     const workerpoolOrderbook = await iexec.orderbook.fetchWorkerpoolOrderbook({
       workerpool: WORKERPOOL_ADDRESS,
-      app: appAddress,
+      app: app,
```
Comment (cedric25) [inline packages/sdk/src/lib/dataProtectorSharing/consumeProtectedData.ts:70]:
```suggestion
      app,
```

File: packages/sdk/src/lib/dataProtectorSharing/smartContract/preflightChecks.ts [hunk 2]
```diff
@@ -229,3 +244,93 @@ export const onlyBalanceNotEmpty = async ({
...
+    BigInt(GROUP_MEMBER_PURPOSE)
+  );
+  if (!isRegistered) {
+    throw new Error(
+      `This whitelist contract does not have registered this app: ${app}.`
+    );
+  }
+};
+
+export const onlyAppOwnBySharingContract = async ({
```
Comment (cedric25) [inline packages/sdk/src/lib/dataProtectorSharing/smartContract/preflightChecks.ts:312]:
```suggestion
export const onlyAppOwnedBySharingContract = async ({
```

File: packages/sdk/src/lib/dataProtectorSharing/smartContract/preflightChecks.ts [hunk 3]
```diff
@@ -229,3 +244,93 @@ export const onlyBalanceNotEmpty = async ({
...
+  sharingContractAddress,
+  pocoAppRegistryContract,
+  app,
+}: {
+  sharingContractAddress: Address;
+  pocoAppRegistryContract: IRegistry;
+  app: Address;
+}) => {
+  const appTokenId = getBigInt(app).toString();
+  console.log(appTokenId, app);
```
Comment (cedric25) [inline packages/sdk/src/lib/dataProtectorSharing/smartContract/preflightChecks.ts:322]: Clean debug log?

File: packages/sdk/src/lib/dataProtectorSharing/subscribeToCollection.ts [hunk 4]
```diff
@@ -41,10 +46,14 @@ export const subscribeToCollection = async ({
 
   try {
     const { txOptions } = await iexec.config.resolveContractsClient();
-    const tx = await sharingContract.subscribeTo(vCollectionTokenId, {
-      value: collectionDetails.subscriptionParams.price,
-      ...txOptions,
-    });
+    const tx = await sharingContract.subscribeTo(
+      vCollectionTokenId,
+      vDuration,
```
Comment (cedric25) [inline packages/sdk/src/lib/dataProtectorSharing/subscribeToCollection.ts:51]: Why do we need to pass this `duration` here?
The smart-contract should know the duration, as it has been set by the collection owner?

**Review by cedric25 — APPROVED** [review]
Quick review, some minor comments + one question

File: packages/sdk/src/lib/dataProtectorSharing/subscribeToCollection.ts (hunk 4 above)
Comment (Le-Caignec) [reply packages/sdk/src/lib/dataProtectorSharing/subscribeToCollection.ts:51]: in order to avoid front run

**Pull request merged**